File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        reverse(begin(nums),end(nums));
        int n = nums.size(); 
        k=k>n?k%n:k; 
        reverse(begin(nums),begin(nums)+k); 
        reverse(begin(nums)+k,end(nums));

    }
};

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k = k%len(nums) if k>=len(nums) else k
        nums[:]=reversed(nums)
        nums[:k]=reversed(nums[:k])
        nums[k:]=reversed(nums[k:]) 

        

        